README.md

```diff
@@ -1,31 +1,208 @@
+
+If outside the container, running `payu setup` command calls the launcher script (e.g. `apps/conda_scripts/payu-1.1.6.d/bin/payu`), which runs a `singularity exec` command that uses the base container in `apps/base_conda/etc/base.sif` and adds the overlay of the SquashFS file `apps/base_conda/envs/payu-1.1.6.sqsh`. The SquashFS is a read-only file system that is mounted to the base container at runtime. This means all the files in `/opt/conda/payu-1.1.6` are now accessible inside the container. The launcher script then runs `payu setup` command inside the container, (e.g. `/opt/conda/payu-1.1.6/bin/payu`).
+
+If the `payu setup` launcher script is invoked inside the container, it will instead run the `payu setup` command directly.
+
+## Github Workflows Overview
+
+General overview of the bash scripts used in the Github workflows:
+
+- **Build script** (`scripts/build.sh`)
+  - Launches the base singularity container and mounts `/g` directory to a temporary directory on `/jobfs` so it does not affect the production environments but inside the container, it is as-if building directly to the deployed locations. Then:
+    - installs a base micromamba, if it doesn't already exist.
+    - builds the micromamba environment
+    - creates all the launcher scripts
+    - sets up the modulefiles
+    - runs any custom environment `build_inner.sh` scripts
+  - Creates a SquashFS file of the environment and stages this file (in `$ADMIN_DIR/staging`).
+  - Once the build is done, archives the environment components into a tar file and stages this file.
+
+- **Test Script** (`scripts/test.sh`)
+  - Unpacks the archived tar file of the built environment components to a temporary location.
+  - Launches the container with the built SquashFS overlay.
+  - Inside the container, run pytests that attempt to import all accessible Python modules. This uses settings defined in an environment's `testconfig.yml` file.
+
+- **Deploy Script** (`scripts/deploy.sh`)
+  - Unpacks the archived tar file of the built environment components to a temporary location.
+  - Rsyncs the apps and modules directories, and moves the SquashFS file to the deployed locations.
+  - Stores the archived tar file in the admin directory.
+  - Runs any custom environment `deploy.sh` scripts.
+
+### Pull Request Workflow (`pull_request.yml`)
+
+On a Pull Request to the `main` branch of this repository, it will check for any changes to any changes to conda environment configuration files (under `environments/`). If changes are detected, it will build a base singularity container on a Github runner, and request a signoff to deployment to a Github Environment. Once approved, it will rsync the repository and the built base container to Gadi, then run the build and test scripts which are submitted as PBS jobs.
```

Comment:
```suggestion
On a Pull Request to the `main` branch of this repository, it will check for changes to any conda environment configuration files (under `environments/`). If changes are detected, it will build a base singularity container on a Github runner, and request a signoff to deployment to a Github Environment. Once approved, it will rsync the repository and the built base container to Gadi, then run the build and test scripts which are submitted as PBS jobs.
```